service/raft/raft_group0.cc

```diff
@@ -432,13 +432,35 @@ future<> raft_group0::join_group0(std::vector<gms::inet_address> seeds, bool as_
     group0_log.info("server {} joined group 0 with group id {}", my_id, group0_id);
 }
 
+struct group0_members {
+    const raft::server& _group0_server;
+    const raft_address_map<>& _address_map;
+
+    std::vector<gms::inet_address> get_inet_addrs() const {
```

Comment:
I really don't like the fact that if we're missing some IP addresses, we just return an empty vector and the caller cannot print a useful log message which explains which members had missing addresses.

If, in production, we end up in some situation where we're not able to learn the IP address of some node for whatever reason (a bug or sth), we won't know which node it is. We won't know which node we should try to restart and so on.

Here's what I would suggest doing, if it's not too much of a pain for you:
1. create another struct, say `addresses`, with two fields: `std::vector<gms::inet_address> found` and `std::vector<raft::server_id> missing` (maybe you could find better names if necessary)
2. Instead of `return {}`, we would always process all members. For those which we cannot translate, put them in `missing`
3. On the call sites, check if `!missing.empty()` - if yes, log the missing members and retry.

Reply:
It won't be enough because it wont' explain why some members are missing - to track down such issues we need the event chain including gossiper events and configuration change updates. I had it all covered with debug logging in state_change_subscriber and raft_address_map API implementations,  because I indeed spent a lot of time debugging the event flow when writing this patch. You requested me to remove all these.
I really don't like it when some code which is useful  and checked in practice is replaced with code which is only theoretically useful.

Reply:
Improved the logging.